packages/flutter_tools/test/integration/flutter_run_test.dart

```diff
@@ -53,6 +53,7 @@ void main() {
       final File pidFile = tempDir.childFile('test.pid');
       await _flutter.run(pidFile: pidFile);
       expect(pidFile.existsSync(), isTrue);
+      await _flutter?.stop();
```

Comment:
Good spot!

Do we need to ensure this runs even if the line above throws though?

Reply:
If there is no pidFile, is there a process running? `_flutter.stop()` doesn't like being called when process is not running(like in test above).

Reply:
If the process runs but doesn't create the pid-file (which is the bug that resulted in this test), the process would've still been created so the `stop()` method should be good to call (it skips most of its code if the debugger never attached/app never started and just waits on `_proc.exitCode`).

Reply:
Okay, got you.